### DISCUSSION TITLE
Use Foundry better [2]

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#2285] [Ideas]
GM @PatrickAlphaC 
at Foundry101 Course, Section 2, EP 14 (Foundry tests cheatcodes), you created 2 new view functions to get some value and use the function in the test files :
![image](https://github.com/user-attachments/assets/c4ba73fe-1f48-4f86-8261-32b102d55c5e)
<br>
![image](https://github.com/user-attachments/assets/5d69991e-6aec-4733-9a60-fd0cc6ec7c1a)
***
it's OK and good but i made some other way without declaring new functions 👀 
![image](https://github.com/user-attachments/assets/e127fbf4-b566-4a37-b361-f2a4bc160d93)
<br>
![image](https://github.com/user-attachments/assets/b7506ae9-ba71-4753-a8a3-56dd6937a7fe)
<br>
![image](https://github.com/user-attachments/assets/40c4da93-6f84-4540-945b-e3d7caaaae1a)
***
the syntax: `CONTRACT.FUNCTION_NAME(INPUTS)` and it will automaticly return the value.
***
And You can Use `ether` keyword beacuse it's `10 ** 18`.
***
GN

**Comment by EngrPips** [comment]
Interesting, You are crushing this thing!

**Reply by MostPow3rful** [reply]
Thanks Sir

**Reply by EngrPips** [reply]
Anytime sir.

**Comment by MostPow3rful** [comment]
![image](https://github.com/user-attachments/assets/38c4cb8b-e36f-4c58-bf63-f09ef908cdb4)
<br>
![image](https://github.com/user-attachments/assets/434142ff-9d0b-4460-ab09-e2e35949fabe)
***
GM Again 👀 You can write this better to :
```js
uint256 startingOwnerBalance = fundMe.i_owner();
```
the `i_owner` is in the FundMe Contract :
```js
address public i_owner;
```
***
and before all of this, i wrote a test for the owner too. :
![image](https://github.com/user-attachments/assets/8457931b-d65e-4548-af13-eca4af8db9ce)
<br>
![image](https://github.com/user-attachments/assets/68da9e6f-89b3-4e32-a1fd-278e6e93e3dc)
***
GN

**Reply by EngrPips** [reply]
I think Patrick explained the motive behind making most variables private and creating a getter for them. You know it isn't really nice to do `fundMe.i_owner(),` but it is fancier and looks nice to do `fundMe.getOwner()` plus we want to default to make all our variables private and only set up a getter for any of them that need to have a getter.

**Reply by MostPow3rful** [reply]
yes i agree and i don't mean he is wrong or something like this. i mean it's better to know this other methods too and in some cases we don't have any getter function, so we should read the variables like this (public) or read from the storage (private). this senarios maybe be in CTFs/challenges or some codebases. but at all, you are right and i am agree with you sir

**Reply by EngrPips** [reply]
I get the gist, sir, and you are crushing it!

**Comment by PatrickAlphaC** [comment]
Yep, we teach both! I think for now we want to keep the code as is. What do you think?

**Reply by MostPow3rful** [reply]
It's OK but it's better to create a good mindset for people and teach them all the ways/methods

**Reply by EngrPips** [reply]
Interesting
